### PR TITLE
Example file consistency

### PR DESCRIPTION
**Pull request by ypar**
In this PR, I am primarily doing some reorganizations to prepare cimr for additional features.

* Additional option for yaml files.
    - 'context_variable_type' has been added as a new option in the yaml file
    - catalog.txt header in defaults and expected_catalog.txt files have been updated to accomodate.

* Example directory has been changed for consistency
  - eqtl and gwas datatype tests are now in its own directories (as datatypes to test will increase in the next cimr release)
  - previously, tgz test used md5 hash and other tests used diff for catalog.txt verification. now all tests will use their respective expected_catalog.txt for diff verifications
  - previously, dbox test did not verify catalog.txt, now it will.

* Added comments for external contributors.
It will continue to be improved, but I have first added more comments for defaults and yamler.

### CODE REVIEW

File: .circleci/test_data_type.sh [hunk 1]
```diff
@@ -2,10 +2,14 @@
 
 set -e -x
 
-cd examples/eqtl_gwas/
+cd examples/
```
Comment (dongbohu) [inline .circleci/test_data_type.sh]: I just realized that the absolute path (`~/cimr/examples/`) will be better than relative path. When using the former, we never need to worry about from where the current script is executed. For example, when using absolute path, `test_data_type.sh` can be launched by either:
```shell
cd ~/cimr/circleci
bash ./test_data_type.sh
```
or:
```shell
bash .circleci/test_data_type.sh  # assume that we are in ~/cimr/
```
But with the relative path, this script will have to be executed ONLY when the current working directory is `~/cimr/`. Does this make sense?

File: .circleci/test_dbox.sh [hunk 2]
```diff
@@ -2,14 +2,19 @@
 
 set -ex
 
-cd ~/cimr/examples/dbox/
+cd examples/dbox/
```
Comment (dongbohu) [inline .circleci/test_dbox.sh]: See my comments on absolute path.

File: .circleci/test_tgz.sh [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 set -ex
 
-cd ~/cimr/examples/tgz/
+cd examples/tgz/
```
Comment (dongbohu) [inline .circleci/test_tgz.sh]: See my comments on absolute path.

File: cimr/processor/processor_prompt.py [hunk 4]
```diff
@@ -149,7 +149,16 @@ def processor_cli(args):
 
         if args.yaml_file:
             yaml_file = [pathlib.Path(args.yaml_file),]
-            y, genome_build, fileset, columnset = convert_yaml(yaml_file)
+            y, fileset = convert_yaml(yaml_file)
+
+            # Check Yamler for required arguments
+            if hasattr(y, 'columnset'):
```
Comment (dongbohu) [inline cimr/processor/processor_prompt.py]: If you want, lines 155-158 can be simplified into:
```python
columnset = getattr(y, 'columnset', {})
```

File: cimr/processor/processor_prompt.py [hunk 5]
```diff
@@ -149,7 +149,16 @@ def processor_cli(args):
...
+                columnset = y.columnset
+            else:
+                columnset = {}
+            if hasattr(y, 'genome_build'):
```
Comment (dongbohu) [inline cimr/processor/processor_prompt.py]: Don't you need a default value for `genoe_build` too (just as you did with `columnset`)? Without the default value, if `y.genome_build` doesn't exist, line 176 would complain that the variable is not defined.

File: cimr/processor/yamler.py [hunk 6]
```diff
@@ -425,24 +435,34 @@ def download(self):
 
         if not os.path.isfile(self.sub_datatype_dir + self.infile):
             logging.info(f' starting download')
+            # Identifiable partial strings in links are used
+            # to prompt different download functions
+
+            # Google drive file links are contain hashed IDs
```
Comment (dongbohu) [inline cimr/processor/yamler.py]: `are` seems redundant.

File: .circleci/test_data_type.sh (hunk 1 above)
Comment (ypar) [reply .circleci/test_data_type.sh]: Sure, it can be changed.

File: cimr/processor/processor_prompt.py (hunk 5 above)
Comment (ypar) [reply cimr/processor/processor_prompt.py]: submission without genome_build should have exited at `convert_yaml()`

Comment (dongbohu) [reply cimr/processor/processor_prompt.py]: So lines 159-160 can be simply written into:
```
genome_build = y.genome_build
```
right?

Comment (ypar) [reply cimr/processor/processor_prompt.py]: sure

**Pull request merged**